.buildkite/cache-builder.yml

```
@@ -0,0 +1,46 @@
+# This script is run via Buildkite's scheduled jobs feature.
+#
+# It's meant to rebuild various CI caches on a periodic async basis, so as
+# not to waste time on every CI job updating the cache.
+
+# Nodes with values to reuse in the pipeline.
+common_params:
+  # Common plugin settings to use with the `plugins` key.
+  - &common_plugins
+    - automattic/bash-cache#v1.4.0
+    - automattic/git-s3-cache#v1.1.0:
+        bucket: "a8c-repo-mirrors"
+        repo: "wordpress-mobile/wordpress-ios/"
+  # Common environment values to use with the `env` key.
+  - &common_env
+    IMAGE_ID: xcode-12.5.1
+
+steps:
+
+  #################
+  # Build the CocoaPods Base Cache
+  # 
+  # This prevents the base cache from infinite growth caused by storing every
+  # version of every pod we've ever used.
+  #################
+  - label: ":cocoapods: Rebuild CocoaPods cache"
+    command: |
+      echo "--- :rubygems: Setting up Gems"
+      install_gems
+
+      echo "--- :cocoapods: Rebuilding Pod Cache"
+      cache_cocoapods
+    env: *common_env
+    plugins: *common_plugins
+
+  #################
+  # Build the Git Repo cache
+  #
+  # Because this repo is so large, we create a Git Mirror and copy it to S3, where
+  # it can be fetch by agents more quickly than from GitHub.
```

```suggestion
  # Because this repo is so large, we periodically create a Git Mirror and copy it to S3,
  # from where it can be fetched by agents more quickly than from GitHub, and so that
  # agents then have less new commits to `git pull` by using that cache as starting point
```
Just thought it could help clarify things to mention that even if we fetch the initial Git Repo via the mirror that is build overnight, we still pull the latest commits on top of that mirror/cache afterwards – otherwise people might wonder when reading this if there isn't a risk that we might be missing commits (from the mirror being too old after a couple of hours and new commits) when we run tests on new PRs 😉

---

Ya for sure! Can never over-document it, right? 😁